services/dictionary-api/src/main/java/datawave/webservice/results/datadictionary/DefaultDataDictionary.java

```diff
@@ -235,7 +235,7 @@ public String getMainContent() {
             builder.append("<td>").append(f.isIndexOnly()).append("</td>");
             builder.append("<td>").append(f.isForwardIndexed() ? true : "").append("</td>");
             builder.append("<td>").append(f.isReverseIndexed() ? true : "").append("</td>");
-            builder.append("<td>").append(f.isNormalized() ? true : "").append("</td>");
+            builder.append("<td>").append(f.getTypes() != null && f.getTypes().size() > 0 ? "true" : "false").append("</td>");
```

Comment:
This looks like it'll work to me. I'm happy to approve if others agree. One thing I'm considering: if the normalized flag is unreliable, should we push this logic into DefaultMetadataField as the definition of `isNormalized`? We could drop the setter for it.

Reply:
I think we have to define the purpose of this column. With the change as-is, I think everything would get listed as normalized since basically everything gets LcNoDiacriticsType.

Reply:
Brian.  It is about 50% true, 50% false for normalized in the Quickstart datadictionary.


FieldName | Internal FieldName | DataType | Index only | Forward Indexed | Reverse Indexed | Normalized | Types | Description | LastUpdated
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
BAR_FIELD |   | csv | false | true | true | true | Text |   | 20160425200000
CONTENT_URI |   | csv | false | true | true | true | Text |   | 20160425200000
CONTRIBUTOR_ID |   | enwiki | false | true |   | true | Number |   | 20130304190000
CONTRIBUTOR_USERNAME |   | enwiki | false | true | true | true | Text |   | 20130304190000
EDGE_VERTEX_FROM |   | csv | false | true | true | true | Text |   | 20160425200000
EDGE_VERTEX_TO |   | csv | false | true | true | true | Text |   | 20160425200000
EMBEDDED_CAST_CHARACTER_ID |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_CHARACTER_IMAGE_MEDIUM |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_CHARACTER_IMAGE_ORIGINAL |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_CHARACTER_LINKS_SELF_HREF |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_CHARACTER_NAME |   | tvmaze | false | true |   | true | Text |   | 20190725152734
EMBEDDED_CAST_CHARACTER_NAME_TOKEN |   | tvmaze | true | true |   | false |   |   | 20190725152734
EMBEDDED_CAST_CHARACTER_URL |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_PERSON_ID |   | tvmaze | false | true |   | true | Text |   | 20190725152734
EMBEDDED_CAST_PERSON_IMAGE_MEDIUM |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_PERSON_IMAGE_ORIGINAL |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_PERSON_LINKS_SELF_HREF |   | tvmaze | false |   |   | false |   |   | 20190725152734
EMBEDDED_CAST_PERSON_NAME |   | tvmaze | false | true |   | true | Text |   | 20190725152734
EMBEDDED_CAST_PERSON_NAME_TOKEN |   | tvmaze | true | true |   | false |   |   | 20190725152734
EMBEDDED_CAST_PERSON_URL |   | tvmaze | false |   |   | false |   |   | 20190725152734
EVENT_DATE |   | csv | false | true |   | true | Text |   | 20160425200000
EVENT_ID |   | csv | false | true | true | true | Text |   | 20160425200000
EXTERNALS_IMDB |   | tvmaze | false | true |   | true | Text |   | 20190725152734
EXTERNALS_THETVDB |   | tvmaze | false | true |   | true | Text |   | 20190725152734
EXTERNALS_TVRAGE |   | tvmaze | false | true |   | true | Text |   | 20190725152734
FIELDNAME1_NOT_TO_TOKENIZE |   | csv | false | true | true | true | Text |   | 20160425200000
FIELDNAME1_WE_DONT_WANT_INDEXED |   | csv | false |   |   | false |   |   | 20160425200000
FILE_TYPE |   | csv | false | true | true | true | Text |   | 20160425200000
FOO_FIELD |   | csv | false | true | true | true | Text |   | 20160425200000
GENRES |   | tvmaze | false | true |   | true | Text |   | 20190725152734
HAS_REVISION_COMMENT_CONTENT |   | enwiki | false |   |   | false |   |   | 20130304190000
HAS_REVISION_TEXT_CONTENT |   | enwiki | false |   |   | false |   |   | 20130304190000
ID |   | tvmaze | false | true |   | true | Text |   | 20190725152734
IMAGE_MEDIUM |   | tvmaze | false |   |   | false |   |   | 20190725152734
IMAGE_ORIGINAL |   | tvmaze | false |   |   | false |   |   | 20190725152734
LANGUAGE |   | tvmaze | false |   |   | false |   |   | 20190725152734
LANGUAGE |   | enwiki | false |   |   | false |   |   | 20130304190000
LANGUAGE |   | csv | false | true | true | true | Text |   | 20160425200000
LINKS_NEXTEPISODE_HREF |   | tvmaze | false |   |   | false |   |   | 20190725152734
LINKS_PREVIOUSEPISODE_HREF |   | tvmaze | false |   |   | false |   |   | 20190725152734
LINKS_SELF_HREF |   | tvmaze | false |   |   | false |   |   | 20190725152734
LOAD_DATE |   | tvmaze | false |   |   | false |   |   | 20190725152734
LOAD_DATE |   | enwiki | false |   |   | false |   |   | 20130304190000
LOAD_DATE |   | csv | false | true | true | true | Text |   | 20160425200000
MD5 |   | csv | false | true | true | true | Text |   | 20160425200000
MY_DATE |   | csv | false | true | true | true | Unknown |   | 20160425200000
NAME |   | tvmaze | false | true | true | true | Text |   | 20190725152734
NAME_TOKEN |   | tvmaze | true | true |   | false |   |   | 20190725152734
NETWORK_COUNTRY_CODE |   | tvmaze | false |   |   | false |   |   | 20190725152734
NETWORK_COUNTRY_NAME |   | tvmaze | false |   |   | false |   |   | 20190725152734
NETWORK_COUNTRY_TIMEZONE |   | tvmaze | false |   |   | false |   |   | 20190725152734
NETWORK_ID |   | tvmaze | false |   |   | false |   |   | 20190725152734
NETWORK_NAME |   | tvmaze | false | true | true | true | Text |   | 20190725152734
NETWORK_NAME_TOKEN |   | tvmaze | true | true |   | false |   |   | 20190725152734
OFFICIALSITE |   | tvmaze | false | true | true | true | Text |   | 20190725152734
ORIG_FILE |   | tvmaze | false |   |   | false |   |   | 20190725152734
ORIG_FILE |   | enwiki | false |   |   | false |   |   | 20130304190000
ORIG_FILE |   | csv | false | true | true | true | Text |   | 20160425200000
ORIGINAL_SIZE |   | csv | false | true | true | true | Text |   | 20160425200000
PAGE_ID |   | enwiki | false | true |   | true | Number |   | 20130304190000
PAGE_NAMESPACE |   | enwiki | false | true |   | true | Number |   | 20130304190000
PAGE_REDIRECT_TITLE |   | enwiki | false |   |   | false |   |   | 20130304190000
PAGE_TITLE |   | enwiki | false | true | true | true | Text |   | 20130304190000
PARAGRAPH_CHECKSUM_SHA_1 |   | csv | false | true | true | true | Text |   | 20160425200000
PARENT_FILETYPE |   | csv | false | true | true | true | Text |   | 20160425200000
PREMIERED |   | tvmaze | false |   |   | true | Unknown |   | 20190725152734
PROCESSED_SIZE |   | csv | false | true | true | true | Text |   | 20160425200000
PROCESSING_DATE |   | csv | false | true |   | true | Text |   | 20160425200000
R1_FIELD1 |   | csv | false | true | true | true | Text |   | 20160425200000
R1_FIELD2 |   | csv | false | true | true | true | Text |   | 20160425200000
RATING_AVERAGE |   | tvmaze | false |   |   | false |   |   | 20190725152734
REVISION_COMMENT |   | enwiki | false |   |   | false |   |   | 20130304190000
REVISION_COMMENT_TERM_COUNT |   | enwiki | false | true |   | true | Number |   | 20130304190000
REVISION_COMMENT_TOKEN |   | enwiki | true | true | true | true | Text |   | 20130304190000
REVISION_FORMAT |   | enwiki | false |   |   | false |   |   | 20130304190000
REVISION_ID |   | enwiki | false | true |   | true | Number |   | 20130304190000
REVISION_MODEL |   | enwiki | false |   |   | false |   |   | 20130304190000
REVISION_PARENTID |   | enwiki | false | true |   | true | Number |   | 20130304190000
REVISION_SHA1 |   | enwiki | false | true |   | true | Text |   | 20130304190000
REVISION_TEXT_SPACE |   | enwiki | false |   |   | false |   |   | 20130304190000
REVISION_TEXT_TERM_COUNT |   | enwiki | false | true |   | true | Number |   | 20130304190000
REVISION_TEXT_TOKEN |   | enwiki | true | true | true | true | Text |   | 20130304190000
REVISION_TIMESTAMP |   | enwiki | false | true |   | true | Text |   | 20130304190000
RUNTIME |   | tvmaze | false | true |   | true | Number |   | 20190725152734
SCHEDULE_DAYS |   | tvmaze | false |   |   | false |   |   | 20190725152734
SCHEDULE_TIME |   | tvmaze | false |   |   | false |   |   | 20190725152734
SECURITY_MARKING |   | csv | false | true |   | true | Text |   | 20160425200000
SHA1 |   | csv | false | true | true | true | Text |   | 20160425200000
SHA256 |   | csv | false | true | true | true | Text |   | 20160425200000
STATUS |   | tvmaze | false | true |   | true | Text |   | 20190725152734
SUMMARY |   | tvmaze | false |   |   | false |   |   | 20190725152734
SUMMARY |   | csv | false | true | true | true | Text |   | 20160425200000
SUMMARY_TOKEN |   | tvmaze | true | true |   | false |   |   | 20190725152734
TYPE |   | tvmaze | false | true |   | true | Text |   | 20190725152734
UPDATED |   | tvmaze | false |   |   | false |   |   | 20190725152734
URL |   | tvmaze | false | true | true | true | Text |   | 20190725152734
UUID |   | csv | false | true | true | true | Text |   | 20160425200000
WEBCHANNEL_ID |   | tvmaze | false |   |   | false |   |   | 20190725152733
WEBCHANNEL_NAME |   | tvmaze | false |   |   | false |   |   | 20190725152733
WEIGHT |   | tvmaze | false |   |   | true | Number |   | 20190725152734

Reply:
I think this is actually correct.  I was surprised that some of the normalized=false columns were on fields that were indexed, however this was on the tokenized fields which are actually normalized if you look at the data in the index.  However the configuration does not reflect that.  I believe I always specify a type on the token fields so this would not be an issue in my use of this code.